itests/hive-unit/src/test/java/org/apache/hadoop/hive/metastore/TestMetastoreHousekeepingLeader.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hive.metastore;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+/**
+ * Test for specifying a valid hostname as HMS leader.
+ */
+public class TestMetastoreHousekeepingLeader extends MetastoreHousekeepingLeaderTestBase {
+  private static final Logger LOG = LoggerFactory.getLogger(TestMetastoreHousekeepingLeader.class);
+
+  @Before
+  public void setUp() throws Exception {
+    internalSetup("localhost");
```

Comment:
insertnal stop can be added to stop the metastore. This way the test can be added in one test file with no duplicate code.

Reply:
In a test, a metastore is started within the same JVM so just stopping the metastore doesn't stop all the threads it runs. Junit takes care of that when running different tests. So, if we add all the scenarios in the same test, we wouldn't know whether a housekeeping thread was started by previous test invocation or current test invocation. Basically a bug in stopMetastore() would be masked. It's better to not go that route. Having separate tests for each scenario is much cleaner.